proxy/http/HttpSM.cc

```diff
@@ -1335,6 +1372,42 @@ HttpSM::state_api_callout(int event, void *data)
     pending_action = nullptr;
   // FALLTHROUGH
   case EVENT_NONE:
+    if (cur_hook_id == TS_HTTP_TXN_START_HOOK && t_state.client_info.port_attribute == HttpProxyPort::TRANSPORT_BLIND_TUNNEL) {
+      /* Creating the request object early to set the host header and port for blind tunneling here for the
+plugins required to work with sni_routing.
+*/
+      // Plugins triggered on txn_start_hook will get the host and port at that point
+      // We've received a request on a port which we blind forward
+      URL u;
+
+      t_state.hdr_info.client_request.create(HTTP_TYPE_REQUEST);
```

Comment:
Isn't this `setup_blind_tunnel_port` again? This should be consolidated, not cut and pasted.

Reply:
well , not quite. `if (!t_state.hdr_info.client_request.url_get()->host_get(&host_len))` this checking in `setup_blind_tunnel_port` is required to avoid setting the hosts again in case of plugin interference at the start hook.

Reply:
Weak. This code is extremely similar, it will need to be cleaned up later. Look at 1390-1401 vs. 553-563. And 1379-1383 vs. 546-550.